leakcanary-deobfuscation-gradle-plugin/src/test/java/com/squareup/leakcanary/deobfuscation/LeakCanaryLeakDeobfuscationPluginTest.kt

```diff
@@ -42,7 +42,7 @@ class LeakCanaryLeakDeobfuscationPluginTest {
     )
   }
 
-  @Test
+  /*@Test
```

Comment:
@mzgreen any idea what could be happening here? With the gradle 4.2 update, the `leakcanary deobfuscation plugin runs and copies mapping file into the apk assets dir` test fails, obfuscationMappingEntry is null.

Reply:
The `leakCanaryObfuscationMapping.txt` is copied to the merged asset folder, but that doesn't get zipped into the APK.

Reply:
Looks like something in AGP has been changed. Maybe it has something to do with the new [resource compiler](https://developer.android.com/studio/releases/gradle-plugin#resource-compiler)? The doc only says that it's enabled by default, not sure if there is a flag allowing to turn it off to check if that's the issue. But anyways even if it's because the new resource compiler it would be better to fix it. Not sure what is going on under the hoods unfortunately :/

Reply:
I'm looking at the test now, it writes a gradle build file [here](https://github.com/square/leakcanary/blob/1d93c749db8991f99af9cd5bec7aeb8702a9f053/leakcanary-deobfuscation-gradle-plugin/src/test/java/com/squareup/leakcanary/deobfuscation/LeakCanaryLeakDeobfuscationPluginTest.kt#L47). have you tried to update this build script to gradle 4.2 like other scripts in the project? I mean replacing `compileSdkVersion` with `compileSdk` etc?

Reply:
I did.

Reply:
Here's a branch that reproduces the failure and has the build scripts updated: https://github.com/square/leakcanary/commits/py/agp_update

Reply:
I pulled it and enabled the plugin on sample app and it looks like it copied the mapping, so maybe the issue is only about the test? 🤔

Reply:
Ok I think I've found a solution. I checked what tasks are executed during the build. leak canary task which copies the mapping file is executed after mapping file generating task is finished and then the app is packaged. I noticed that `merge{Variant}Assets` task is executed earlier and that's fine. But after `merge{Variant}Assets` there is another task called `compress{Variant}Assets`. So I did an experiment and hooked up the tasks so that the `compress{Variant}Assets` task is executed after leak canary task and it turns out that now the assets directory shows up in the apk. I'm still not sure why it works like this but it seems that this compression step is important. 

Now to fix the issue you can open `LeakCanaryLeakDeobfuscationPlugin` file and paste this
```
    findTaskProviderOrNull(
      project,
      "compress${variant.name.capitalize()}Assets"
    )?.configure {
      it.dependsOn(copyObfuscationMappingFileTaskProvider)
    }
```
in line number 98.

[Here](https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:build-system/gradle-core/src/main/java/com/android/build/gradle/internal/tasks/CompressAssetsTask.kt) is the source of `CompressAssetsTask` but it's the newest one, not sure how to find the version from AGP 4.2.2.